runner/android_junit_runner/javatests/androidx/test/internal/runner/TestRequestBuilderTest.java

```diff
@@ -227,6 +227,50 @@ public String describe() {
     }
   }
 
+  @SampleCustomAnnotationOnClass(runTests = true)
+  public static class SampleCustomFilterOnClassRunTestsTestClass {
+    @Test
+    public void testOneRun() {}
+
+    @Test
+    public void testTwoRun() {}
+  }
+
+  @SampleCustomAnnotationOnClass(runTests = false)
+  public static class SampleCustomFilterOnClassSkipTestsTestClass {
+    @Test
+    public void testOneSkip() {}
+
+    @Test
+    public void testTwoSkip() {}
+  }
+
+  public static class SampleCustomFilterOnClass extends AbstractFilter {
+    public SampleCustomFilterOnClass() {}
+
+    @Override
+    public boolean shouldRun(Description description) {
+      return evaluateTest(description);
+    }
+
+    @Override
+    protected boolean evaluateTest(Description description) {
+      return description.getAnnotation(SampleCustomAnnotationOnClass.class).runTests();
+    }
+
+    @Override
+    public String describe() {
+      return "skip all tests in class if runTests is false";
+    }
+  }
+
+  @Retention(RetentionPolicy.RUNTIME)
+  @Target({ElementType.TYPE})
+  @CustomFilter(filterClass = SampleCustomFilterOnClass.class)
+  public @interface SampleCustomAnnotationOnClass {
```

Comment:
This annotation shouldn't exist, it's essentially the same as `SampleCustomAnnotation`, just below.